src/Raven.Server/Documents/Handlers/Debugging/DebugPackage/Analyzers/Database/StorageReportInfoAnalyzer.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System.IO;
+using Raven.Client.Util;
+using Raven.Server.Documents.Handlers.Debugging.DebugPackage.Analyzers.Errors;
+using Raven.Server.Documents.Handlers.Debugging.DebugPackage.Analyzers.Issues;
+using Voron.Debugging;
+
+namespace Raven.Server.Documents.Handlers.Debugging.DebugPackage.Analyzers.Database;
+
+public class StorageReportInfoAnalyzer(
+    string databaseName,
+    DebugPackageAnalyzeErrors errors,
+    DebugPackageAnalysisIssues issues) : AbstractDebugPackageDatabaseAnalyzer(databaseName, errors, issues)
+{
+    public EnvironmentStorageReport StorageReport { get; set; }
+    
+    protected override bool RetrieveAnalyzerInfo(DebugPackageEntries entries)
+    {
+        if (entries.TryGetValue<StorageHandler, EnvironmentStorageReport>(x => x.Report(), out var storageReport) == false)
+        {
+            AddWarning("Failed to get storage report");
+        }
+        
+        StorageReport = storageReport;
+
+        return true;
+    }
+
+    protected override void DetectIssues(DebugPackageAnalysisIssues issues)
+    {
+        foreach (var item in StorageReport.Results)
+        {
+            var report = item.Report;
+
+            long dataFileSizeInBytes = report.DataFile.AllocatedSpaceInBytes;
+            if (dataFileSizeInBytes < Size.GB / 2)
+            {
+                // skip small storages
+                continue;
+            }
+            
+            double freeSpacePercentage = 1.0 * report.DataFile.FreeSpaceInBytes / dataFileSizeInBytes;
+
+            var freeSpaceThreshold = dataFileSizeInBytes > 8 * Size.GB ? 0.5 : 0.6;
+            
+            if (freeSpacePercentage > freeSpaceThreshold)
```

Comment:
This needs to also take into account that we may have 8.0 with the sparse file handling

Reply:
https://issues.hibernatingrhinos.com/issue/RavenDB-24756/Debug-package-analyzer-Space-analysis-need-to-take-into-account-sparse-files-in-8.0